### PR TITLE
ReflectorForSSRPass: expose .color

### PR DESCRIPTION
**Pull request by ycw**
expose `.color` on reflector.

```js
// 1
groundReflector.color.set( 'red' )
// 2
groundReflector.color = 'red' 
```

### CODE REVIEW

File: examples/jsm/objects/ReflectorForSSRPass.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ var Reflector = function ( geometry, options ) {
 	var textureHeight = options.textureHeight || 512;
 	var clipBias = options.clipBias || 0;
 	var shader = options.shader || Reflector.ReflectorShader;
-	var useDepthTexture = options.useDepthTexture
+	var useDepthTexture = options.useDepthTexture === true;
```
Comment (Mugen87) [inline examples/jsm/objects/ReflectorForSSRPass.js:35]: I guess you want to rebase the commit so the changes from #21385 do not pop up in the diff.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
How about this instead?

```js
scope.color = scope.material.uniforms[ 'color' ].value;
```

**Comment by gonnavis** [comment]
The main reason I use `defineProperty` elsewhere in SSRPass is for properties of `defines`, because change them need update material. So for performance reason I think it's better to watch the change event instead of setting them in render loop.

I feel the use of `defineProperty` for `color` is a bit overdone, but not too opposed because I guess @ycw 's intention is that the initial code already accept string value such as 'red', and want to keep it.
`var color = ( options.color !== undefined ) ? new Color( options.color ) : new Color( 0x7F7F7F );`

For me, may codes below is best, but will introduce broken change, just for reference.
```
groundReflector = new Reflector( geometry, {
    ...
    color: new THREE.Color(0x888888),
} );
```
```
scope.color = ( options.color !== undefined ) ? options.color : new Color( 0x7F7F7F );
...
this.doRender = function ( renderer, scene, camera ) {
    material.uniforms[ 'color' ].value = scope.color;
```

**Comment by mrdoob** [comment]
But the library usually doesn't do that, you can't do this:

```js
material.color = 'red';
```

**Comment by gonnavis** [comment]
Yes, the codes I suggested should use `groundReflector.color=new THREE.Color('blue')`, can't do `groundReflector.color='blue'`.

But the Initialization codes inherited form `Reflector.js` can accept pure string like `'blue'`.
```
var color = ( options.color !== undefined ) ? new Color( options.color ) : new Color( 0x7F7F7F );
```
If really need expose `color`, should broken-change the Initialization codes, or leave it as is but only accept `groundReflector.color=new THREE.Color('blue')` for subsequent changes?

**Comment by Mugen87** [comment]
> But the Initialization codes inherited form Reflector.js can accept pure string like 'blue'.

Yes since you can use multiple types to create an instance of `THREE.Color`.

@mrdoob refers to the issue that you can't _assign_ a string or numeric value to a color material property. This only works by using `set()`.

**Pull request merged**